HOOKS.md

```diff
@@ -327,11 +329,30 @@ useRenderActivity(
   }) => React.Element
 ): ({
   activity: Activity,
-  timestampClassName: string
+  nextVisibleActivity: Activity
 }) => React.Element
 ```
 
-This function is for rendering an activity and its attachments inside a React element. Because of the parent-child relationship, the caller will need to pass a render function in order for the attachment to create a render function for the activity. When rendering the activity, the caller will need to pass `activity` and `timestampClassName`. This function is a composition of `activityRendererMiddleware`, which is passed as a prop.
+This function is for rendering an activity and its attachments inside a React element. Because of the parent-child relationship, the caller will need to pass a render function in order for the attachment to create a render function for the activity. When rendering the activity, the caller will need to pass `activity` and `nextVisibleActivity`. This function is a composition of `activityRendererMiddleware`, which is passed as a prop.
+
+Because not all activities are rendered, for example, event activities are not rendered. The `nextVisibleActivity` is the pointer to next visible activity and is intended for activity status renderer on grouping timestamps for adjacent activities.
+
+### New in 4.8.0
+
+Previously, we use `timestampClassName` to control if the activity should show timestamp or not. The `timestampClassName` should be add as a `class` attribute the DOM element which contains the timestamp.
+
+Today, we pass `activity` and `nextVisibleActivity` to the middleware, so the `activityRendererMiddleware` make the decision about timestamp visibility. For example, developers can group timestamp based on activity type.
```

Comment:
We'll need to add this to the changelog. Please leave a note in R8 pre-release checklist

Reply:
Added to #2856.